aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -2060,6 +2060,20 @@ bool cpu_equal(const Tensor& self, const Tensor& other) {
   if (!self.is_same_size(other)) {
     return false;
   }
+  // Since the flags like neg/conj should be already handled outside the
+  // TensorIterator, it should be safe to have the following fast path by
+  // ensuring the storage and strides exactly the same.
+  if (self.dtype() == other.dtype()
+      && self.sizes().equals(other.sizes())
+      && self.strides().equals(other.strides())
+      && self.is_alias_of(other)
+      && self.storage_offset() == other.storage_offset()
+      && self.layout() == other.layout()
+      && self.is_neg() == other.is_neg()
+      && self.is_conj() == other.is_conj()) {
```

Comment:
You don't need to test these three, they will have been handled before getting here.

Reply:
Just to be safe, I am a bit concerned in some cases, if users directly this cpu_equal function directly, although this should be rare. Keeping these checks shouldn't hurt, or we concern about the overhead for these calls?